### PR TITLE
[C] Resolve some issues from #1473

### PR DESCRIPTION
**Pull request by zhemant**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- Changed curl version in CMakeLists
- Change CMakeLists to generate the library
- Updated README with how to compile and use generated library
- Modified import statements according to the directory structure
- Add a variable projectName to use in CMakeLists to differentiate generated packages
- updated .gitignore for C generator

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CLibcurlClientCodegen.java [hunk 1]
```diff
@@ -567,6 +573,23 @@ public void setParameterExampleValue(CodegenParameter p) {
...
+    @Override
+    public void preprocessOpenAPI(OpenAPI openAPI) {
+
+            if (openAPI.getInfo() != null) {
+                Info info = openAPI.getInfo();
+
+                setProjectName((escapeText(info.getTitle())));
+            } else {
+                setProjectName(defaultProjectName);
+            }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CLibcurlClientCodegen.java]: Please use 4-space instead for indentation

**Comment by wing328** [comment]
cc @PowerOfCreation

**Comment by wing328** [comment]
@zhemant looks like the C petstore samples have not been regenerated.

**Comment by zhemant** [comment]
@wing328 Yes I haven't pushed the updated sample I was waiting to merge all the PR related to the enhancement list and then updated all samples in one PR. I have almost finished the list. After this PR I will generate another PR with rest fixes and then Samples so that PR will be clean.

**Comment by wing328** [comment]
@zhemant minor suggestion: create a branch for the change as per git best practice.

**Comment by wing328** [comment]
If the C client is regenerated to the same folder, please be reminded that cJSON.c and cJSON.h are generated to a different location so please make sure the corresponding old files under `src` and `include` folder are deleted.

**Comment by wing328** [comment]
@zhemant thanks for the contribution, which has been included in the 4.0.0-beta release: https://twitter.com/oas_generator/status/1079727020374806529.

Happy New Year and looking forward to more collaboration and contributions in 2019!

**Pull request merged**